src/driver/main.cpp

```diff
@@ -0,0 +1,214 @@
+#include "argument_parser.hpp"
+#include "command.hpp"
+#include "verify.hpp"
+#include "perf.hpp"
+
+#include <migraphx/tf.hpp>
+#include <migraphx/onnx.hpp>
+#include <migraphx/stringutils.hpp>
+
+namespace migraphx {
+namespace driver {
+inline namespace MIGRAPHX_INLINE_NS {
+
+struct loader
+{
+    std::string file;
+    std::string type;
+    bool is_nhwc  = true;
+    unsigned trim = 0;
+
+    void parse(argument_parser& ap)
+    {
+        ap(file, {}, ap.metavar("<input file>"));
+        ap(type, {"--onnx"}, ap.help("Load as onnx"), ap.set_value("onnx"));
+        ap(type, {"--tf"}, ap.help("Load as tensorflow"), ap.set_value("tf"));
+        ap(is_nhwc, {"--nhwc"}, ap.help("Treat tensorflow format as nhwc"), ap.set_value(true));
+        ap(is_nhwc, {"--nchw"}, ap.help("Treat tensorflow format as nchw"), ap.set_value(false));
+        ap(trim, {"--trim", "-t"}, ap.help("Trim instructions from the end"));
+    }
+
+    program load()
+    {
+        program p;
+        if(type.empty())
+        {
+            if(ends_with(file, ".onnx"))
+                type = "onnx";
+            else
+                type = "tf";
```

Comment:
tensorflow protobufs default to .pb file extensions, so I think we should do a check on the file for this